File: sentinel-gateway/src/config/env.ts
import dotenv from "dotenv";
import path from "path";
import { logger } from "./logger.js";

/**
 * SERVICE: Sentinel-AI Configuration & Environment Layer
 * DESCRIPTION: Loads, parses, and validates runtime environment flags across system boundaries.
 * STANDARDS: Synchronous initialization patterns, explicit dependency crash validation, production trace safety.
 */

dotenv.config({ path: path.resolve(process.cwd(), "..", ".env") });

function getEnvVariable(key: string): string {
  const value = process.env[key];
  if (!value) {
    logger.fatal({ missingKey: key }, "Critical system configuration initialization failed");
    throw new Error(`Missing required environment variable: ${key}`);
  }
  return value;
}

export const env = {
  NODE_ENV: process.env.NODE_ENV || "development",
  PORT: parseInt(process.env.PORT || "8008", 10),
  REDIS_URL: getEnvVariable("REDIS_URL"),
  MONGO_URI: getEnvVariable("MONGO_URI"),
  MONGO_DB_NAME: getEnvVariable("MONGO_DB_NAME"),
  CLIENT_URL: getEnvVariable("CLIENT_URL"),
};

File: sentinel-worker/src/config/env.py
import os
from pathlib import Path
from dotenv import load_dotenv
ROOT_DIR = Path(__file__).resolve().parent.parent.parent.parent
env_path = ROOT_DIR / ".env"

load_dotenv(dotenv_path=env_path)

print(f"Searching .env at: {env_path}")
print(f"File exists: {env_path.exists()}")
def get_env_variable(key: str, default=None) -> str:
    value = os.getenv(key, default)
    if value is None:
        raise ValueError(f"Missing required environment variable: {key}")
    return value

class Config:
    REDIS_URL = get_env_variable("REDIS_URL")
    MONGO_URI = get_env_variable("MONGO_URI")
    MONGO_DB_NAME = get_env_variable("MONGO_DB_NAME")
    GATEWAY_URL = get_env_variable("GATEWAY_URL")
env = Config()
